packages/nimble-components/src/rich-text/models/tests/markdown-parser.spec.ts

```diff
@@ -923,4 +928,317 @@ describe('Markdown parser', () => {
+                '1234-5678'
+            );
+        });
+
+        it('should show user ID of first grouping pattern when username not found where the pattern has multiple grouping regex', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'user:1.com', displayName: 'username1' },
+                    { key: 'user:2.com', displayName: 'username2' }
+                ],
+                '^user:(.*)\\.(com)'
+            ));
+            await connect();
+            mentionsMap.set(
+                element.mentionInternals.character,
+                element.mentionInternals
+            );
+            const doc = RichTextMarkdownParser.parseMarkdownToDOM(
+                '<user:1234-5678.com>',
+                mentionsMap
+            );
+
+            expect(getTagsFromElement(doc)).toEqual([
+                'P',
+                `${richTextMentionUsersViewTag}`.toUpperCase()
+            ]);
+            expect(getLastChildElementAttribute('mention-label', doc)).toEqual(
+                '1234-5678'
+            );
+        });
+
+        it('should show username where the pattern has multiple grouping regex', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'user:1.com', displayName: 'username1' },
+                    { key: 'user:2.com', displayName: 'username2' }
+                ],
+                '^user:(.*)\\.(com)'
+            ));
+            await connect();
+            mentionsMap.set(
+                element.mentionInternals.character,
+                element.mentionInternals
+            );
+            const doc = RichTextMarkdownParser.parseMarkdownToDOM(
+                '<user:1.com>',
+                mentionsMap
+            );
+
+            expect(getTagsFromElement(doc)).toEqual([
+                'P',
+                `${richTextMentionUsersViewTag}`.toUpperCase()
+            ]);
+            expect(getLastChildElementAttribute('mention-label', doc)).toEqual(
+                'username1'
+            );
+        });
+
+        it('should show user ID in the absence of mapping elements', async () => {
+            ({ element, connect, disconnect } = await setup([], '^user:(.*)'));
+            await connect();
+            mentionsMap.set(
+                element.mentionInternals.character,
+                element.mentionInternals
+            );
+            const doc = RichTextMarkdownParser.parseMarkdownToDOM(
+                '<user:1234-5678>',
+                mentionsMap
+            );
+
+            expect(getTagsFromElement(doc)).toEqual([
+                'P',
+                `${richTextMentionUsersViewTag}`.toUpperCase()
+            ]);
+            expect(getLastChildElementAttribute('mention-label', doc)).toEqual(
+                '1234-5678'
+            );
+        });
+
```

Comment:
```suggestion
```
This doesn't seem materially different from the earlier test case where the user id does not match any of the mapping elements.

Reply:
Removed this test case.